lyluatex.lua

```diff
@@ -230,31 +230,87 @@ function latex.fullpagestyle(style, ppn)
+    for i = from, to, dir do
+        table.insert(result, i)
+    end
+    return result
 end
 
-function latex.includesystems(filename, protrusion, indent, do_compile)
-    local f = io.open(filename..'.tex', 'r')
-    local texoutput = f:read('*a')
-    f:close()
-    if do_compile then  -- new compilation, update output-systems.tex
-        f = io.open(filename..'.count', 'r')
-        local nsystems = tonumber(f:read('*a'))
-        f:close()
-        if nsystems == 1 and indent then
-            warn('Only one system, deactivating indentation.')
-            protrusion = protrusion - indent
+function latex.include_only(range)
+    if range == '' then return false
+    elseif tonumber(range) then return { range } end
+    local result = {}
+    local ranges = range:explode(',')
+    for _, r in pairs(ranges) do
+        local loc_range = latex.parse_num_or_range(r:gsub('^%s', ''):gsub('%s$', ''))
+        if loc_range then
+            for _, v in pairs(loc_range) do table.insert(result, v) end
         end
-        texoutput =
-            [[\ifx\preLilyPondExample\undefined\else\expandafter\preLilyPondExample\fi]]..
-            texoutput:gsub([[\includegraphics{]], string.format(
-                [[\noindent\hspace*{%spt}\includegraphics{%s]],
-                protrusion, dirname(filename)))..
-            [[\ifx\postLilyPondExample\undefined\else\expandafter\postLilyPondExample\fi]]
-        f = io.open(filename..'.tex', 'w')
-        f:write(texoutput)
+    end
+    return result
+end
+
+function latex.includepdf(pdfname, range)
+    local print_only = latex.include_only(range)
+    if not print_only then print_only = '-'
+    else
+        print_only = '{'..table.concat(print_only, ',')..'}'
+    end
+    tex.sprint(string.format(
+        [[\includepdf[pages=%s]{%s}]], print_only, pdfname))
+end
+
+function latex.includesystems(filename, range, protrusion, indent)
+    local print_only
+    if range == '' then  -- no range given, check available systems
+        local f = io.open(filename..'-systems.count', 'r')
+        local nsystems = f:read('*a')
         f:close()
+        print_only = latex.include_only('1-'..nsystems)
+    else
+        print_only = latex.include_only(range)
+    end
+    if #print_only == 1 and print_only[1] == "1" and indent then
+        warn([[
+Only one system, deactivating indentation.]])
+        protrusion = protrusion - indent
+    end
+    local texoutput = ''
+    if ly.pre_lilypond then
+        texoutput = texoutput..'\\preLilyPondExample'
+    end
+    for index, system in pairs(print_only) do
+        texoutput = texoutput..string.format([[
+        \noindent\hspace*{%spt}\includegraphics{%s}
+        ]],
+        protrusion, filename..'-'..system)
+        if ly.between_lilypond then
+            texoutput = texoutput..string.format([[
+            \betweenLilyPondSystem{%s}
+            ]], index)
+        else
+            texoutput = texoutput..'\n\\par\\bigskip'
```

Comment:
I don't really agree with this: the result is good only when `staffsize` is auto-calculated, not when it's far enough from the default (for example `staffsize=16` with a document's fontsize of 12pt).
I think it should:
- either stick to the default `\linebreak` (as in `lilypond-book`);
- or be a `\vpsace` proportional to `staffsize`.